### PR TITLE
chore(docs): configure doc preview for PRs

### PR DESCRIPTION
**Pull request by mikemckiernan**
Complete the configuration for a Sphinx build.

Add the workflow for doc previews in PRs.

### CODE REVIEW

**Comment by mikemckiernan** [comment]
As with the previous two times, I don't anticipate that the `docs-preview-pr` workflow will run.

In terms of a check, I pushed this commit to the `main` branch of my fork and then opened a PR against it.  It seems OK:

<https://github.com/mikemckiernan/core/pull/2>

File: docs/README.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ here](https://nvidia-merlin.github.io/core).
    directory:
 
    ```sh
-   http-server build/html
+   python -m http.server -d build/html
```
Comment (benfred) [inline docs/README.md:34]: the python http.server module isn't the best imo (and I prefer using the node.js version that was previously listed here - it handles concurrent requests much better, and doesn't flake out as much at least for me).

However, If we are going to recommend going with the python http.server, maybe we can ditch the recommendation on line 37 to install the node.js one ?

**Review by benfred — APPROVED** [review]
thanks! looks good

I'll create the gh-pages branch in a second

File: docs/README.md (hunk 1 above)
Comment (mikemckiernan) [reply docs/README.md:34]: Doh!  I missed that.  I just noticed that I wasn't successful running the command so I substituted back in the vanilla Python method.

I'm not sure that my opinion matters for much, but to me, I have a preference to keeping all the dependencies for a project in a distinct venv.  Mixing in node...I guess my pref is to show the simplest way.  More knowledgeable folks like yourself don't have to follow instructions.

I'll remove line 37. Thank you.

**Pull request merged**